Tasks/AzureStorage/main.ts

```diff
@@ -0,0 +1,45 @@
+import path = require("path");
+import * as os from 'os';
+import util = require("util");
+import tl = require("vsts-task-lib/task");
+import msRestAzure = require("azure-arm-rest/azure-arm-common");
+import armStorage = require("azure-arm-rest/azure-arm-storage");
+import blobUtils = require("./operations/AzureBlobUtils");
+
+function run(): Q.Promise<void> {
+    var artifactsDirectory = tl.getVariable('System.ArtifactsDirectory');
+    var azCopyExeLocation: string = path.join(__dirname, 'AzCopy', 'AzCopy.exe');
+
+    var connectedServiceName = tl.getInput('ConnectedServiceName', true);
+    var storageAccountName = tl.getInput('StorageAccountName', true);
+    var containerName = tl.getInput('ContainerName', true);
+    var commonVirtualPath = tl.getInput('CommonVirtualPath', false);
+    if(!commonVirtualPath) {
+        commonVirtualPath = "";
+    }
+    var subscriptionId = tl.getEndpointDataParameter(connectedServiceName, 'subscriptionid', true);
+    var servicePrincipalId: string = tl.getEndpointAuthorizationParameter(connectedServiceName, "serviceprincipalid", false);
+
+    var credentials = getARMCredentials(connectedServiceName);
+    var armStorageClient: armStorage.StorageManagementClient = new armStorage.StorageManagementClient(credentials, subscriptionId);
```

Comment:
these should not be created here. armStorageClient should be created in AzureBlobUtils module and azCopyExeLocation should be created in AzCopyUtils module. Respected modules should know how to initialize their dependency. Just pass credential from main.ts